main.js

```diff
@@ -7,120 +7,146 @@ const { addExtra } = require("puppeteer-extra");
-		secretAccessKey: process.env.AWSSECRETACCESSKEY,
-	});
+    //S3 bucket initialization
+    const s3 = new AWS.S3({
+        accessKeyId: process.env.AWSACCESSKEYID,
+        secretAccessKey: process.env.AWSSECRETACCESSKEY,
+    });
 
-	Puppeteer.use(
-		Recaptcha({
-			provider: { id: "2captcha", token: process.env.RECAPTCHATOKEN },
-		})
-	);
+    const cachedResults = await fetch(
+        "https://mzqsa4noec.execute-api.us-east-1.amazonaws.com/prod"
+    )
+        .then((res) => res.json())
+        .then((unpack) => JSON.parse(unpack.body).results);
 
-	const browser = process.env.DEVELOPMENT
-		? await Puppeteer.launch({
-				executablePath: process.env.CHROMEPATH,
-		  })
-		: await Puppeteer.launch({
-				args: chromium.args,
-				defaultViewport: chromium.defaultViewport,
-				executablePath: await chromium.executablePath,
-				headless: chromium.headless,
-				ignoreHTTPSErrors: true,
-		  });
+    Puppeteer.use(
+        Recaptcha({
+            provider: { id: "2captcha", token: process.env.RECAPTCHATOKEN },
+        })
+    );
 
-	const gatherData = async () => {
-		const results = await Promise.all(
-			scrapers.map((scraper) =>
-				scraper(browser).catch((error) => {
-					//print out the issue but don't fail, this way we still publish updates
-					//for other locations even if this website's scrape doesn't work
-					console.log(error);
-					return null;
-				})
-			)
-		);
-		browser.close();
-		const finalResultsArray = [];
-		for (const result of results) {
-			if (Array.isArray(result)) {
-				finalResultsArray.push(...result);
-			} else if (result) {
-				//ignore nulls
-				finalResultsArray.push(result);
-			}
-		}
+    const browser = process.env.DEVELOPMENT
+        ? await Puppeteer.launch({
+              executablePath: process.env.CHROMEPATH,
+          })
+        : await Puppeteer.launch({
+              args: chromium.args,
+              defaultViewport: chromium.defaultViewport,
+              executablePath: await chromium.executablePath,
+              headless: chromium.headless,
+              ignoreHTTPSErrors: true,
+          });
 
-		const responseJson = {
-			results: finalResultsArray,
-		};
+    const gatherData = async () => {
+        const results = await Promise.all(
+            scrapers.map((scraper) =>
+                scraper(browser).catch((error) => {
+                    //print out the issue but don't fail, this way we still publish updates
+                    //for other locations even if this website's scrape doesn't work
+                    console.log(error);
+                    return null;
+                })
+            )
+        );
+        browser.close();
+        let scrapedResultsArray = [];
+        for (const result of results) {
+            if (Array.isArray(result)) {
+                scrapedResultsArray.push(...result);
+            } else if (result) {
+                //ignore nulls
+                scrapedResultsArray.push(result);
+            }
+        }
```

Comment:
Why are we looping over the results here when we have an arrow function filtering at L50?
Seems like either this should move to the arrow function or we should loop over the scrapers and push each resul in the loop. Maybe there's a promise subtlety I'm missing.